oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/record-requests/components/RequestModal.jsx

```diff
@@ -21,223 +17,108 @@ import { useConfirmDialog } from "../utils/hooks";
-            {requestModalType === REQUEST_TYPE.CREATE &&
-              <CreateRequestModalContent requestType={request} customSubmitHandler={customSubmitHandler} /> ||
-              <RequestModalContent request={request} requestType={requestType} requestModalType={requestModalType} customSubmitHandler={customSubmitHandler} />
-            }
-          </FormikContext.Provider>
-        </Modal.Content>
-        <Modal.Actions>
-          {requestModalType === REQUEST_TYPE.SUBMIT &&
-            <>
-              <Button title={i18next.t("Submit request")} color="blue" icon labelPosition="left" floated="right" {...submitButtonExtraProps}>
-                <Icon name="paper plane" />
-                {i18next.t("Submit")}
-              </Button>
-              <Button title={i18next.t("Cancel request")} onClick={() => confirmAction(REQUEST_TYPE.CANCEL)} negative icon labelPosition="left" floated="left">
-                <Icon name="trash alternate" />
-                {i18next.t("Cancel request")}
-              </Button>
-              {formWillBeRendered && 
-                <Button title={i18next.t("Save drafted request")} onClick={() => sendRequest(REQUEST_TYPE.SAVE)} color="grey" icon labelPosition="left" floated="right">
-                  <Icon name="save" />
-                  {i18next.t("Save")}
+    <FormikProvider value={formik}>
+      <ConfirmModalContextProvider>
+        {({ confirmDialogProps }) =>
+          <>
+            <Modal
+              className="requests-request-modal"
+              as={Dimmer.Dimmable}
+              blurring
+              onClose={onClose}
+              onOpen={openModal}
+              open={isOpen}
+              trigger={trigger || <Button content="Open Modal" />}
+              closeIcon
+              closeOnDocumentClick={false}
+              closeOnDimmerClick={false}
+              role="dialog"
+              aria-labelledby="request-modal-header"
+              aria-describedby="request-modal-desc"
+            >
+              <Dimmer active={isSubmitting}>
+                <Loader inverted size="large" />
+              </Dimmer>
+              <Modal.Header as="h1" id="request-modal-header">{header}</Modal.Header>
+              <Modal.Content>
+                {error &&
+                  <Message negative>
+                    <Message.Header>{i18next.t("Error sending request")}</Message.Header>
+                    <p ref={errorMessageRef}>{error?.message}</p>
+                  </Message>
+                }
+                <ContentComponent request={requestOrRequestType} requestType={requestType} requestModalType={requestModalContentType} onCompletedAction={onSubmit} />
+              </Modal.Content>
+              <Modal.Actions>
+                {actions.map(({ name, component: ActionComponent }) => 
+                  <ActionComponent key={name} request={requestOrRequestType} requestType={requestType} onSubmit={onSubmit} />
```

Comment:
Because not all props are used in the variable `ActionComponent`s. Some Actions need more props, some less. And since some components need to distinguish between `request` and `requestType` respectively, I also include additional prop. In other cases its just small redundancy. Also `request` and `requestType` have very similar interface in terms of `links`, so in many cases it doesn't matter what we use.